test/parallel_api/algorithm/alg.modifying.operations/shift_left_right.pass.cpp

```diff
@@ -86,7 +86,7 @@ struct test_shift
             //copying data to USM buffer
             ::std::copy_n(first, m, ptr.get());
 
-            auto het_res = algo(::std::forward<Policy>(exec), ptr.get(), ptr.get() + m, n);
+            auto het_res = algo(oneapi::dpl::execution::make_device_policy<class USM>(::std::forward<Policy>(exec)), ptr.get(), ptr.get() + m, n);
```

Comment:
Please make sure the following tests pass when `-DONEDPL_USE_UNNAMED_LAMBDA=False`:
- shift_left_right
- inclusive_scan_by_segment
- exclusive_scan_by_segment
- reduce_by_segment

The issues are similar, e.g. the one for `shift_left_right`:
```
In file included from <built-in>:1:
/tmp/shift_left_right-b1aa61.h:355:20: error: redefinition of 'KernelInfo<sycl::detail::__pf_kernel_wrapper<oneapi::dpl::__par_backend_hetero::__parallel_for_kernel<USM>>>'
template <> struct KernelInfo<::sycl::detail::__pf_kernel_wrapper<::oneapi::dpl::__par_backend_hetero::__parallel_for_kernel<USM>>> {
```